roles/confluent.common/tasks/confluent_cli.yml

```diff
@@ -0,0 +1,45 @@
+---
+- name: Confluent CLI system check
+  assert:
+    that: ansible_facts['system']|lower == "linux"
+    fail_msg: "Confluent CLI system not supported"
+    quiet: true
+
+- name: Confluent CLI architecture check
+  assert:
+    that: confluent_cli_goarch[ansible_facts['architecture']] is defined
+    fail_msg: "Confluent CLI architecture not supported"
+    quiet: true
+
+# confluent_cli_checksums_url contains a SHA256 txt checksum file with checksums for all archives
+- name: Confluent CLI get checksums
+  set_fact:
+    confluent_cli_checksums: "{{lookup('url', confluent_cli_checksums_url, wantlist=True)}}"
+  when:
+    - confluent_cli_archive_path is not defined
+# select confluent_cli_archive name in SHA256 txt file and extract the checksum string to confluent_cli_sha256
+- name: Confluent CLI archive checksum
+  set_fact:
+    confluent_cli_sha256: "{{confluent_cli_checksums|select('match','.*'+confluent_cli_archive)|regex_findall('([a-fA-F0-9]+)\\s*.*')|first|string }}"
+  when:
+    - confluent_cli_archive_path is not defined
+
+- name: Confluent CLI download
+  get_url:
+    url: "{{confluent_cli_s3_archive+confluent_cli_archive}}"
+    dest: "/tmp/{{confluent_cli_archive}}"
+    checksum: "sha256:{{confluent_cli_sha256}}"
+  when:
+    - confluent_cli_archive_path is not defined
+
+- name: Confluent CLI set archive path
+  set_fact:
+    confluent_cli_archive_path: "/tmp/{{confluent_cli_archive}}"
+  when:
+    - confluent_cli_archive_path is not defined
+
+- name: Confluent CLI install binary
+  command:
+    cmd: "tar -C {{confluent_cli_path|dirname}} --strip-components=1 -xzf {{confluent_cli_archive_path}} {{confluent_cli_binary}}/{{confluent_cli_binary}}"
```

Comment:
can this be replaced w the unarchive module? https://docs.ansible.com/ansible/latest/collections/ansible/builtin/unarchive_module.html

What does the `--strip-components=1` do?

Reply:
Hi @domenicbove 

I used `tar` command because the `unarchive` module does not offer directly a feature to extract only one file (there is an exclude filter anyway). The archive contains a folder when extracted, and the binary is inside the folder. The `--strip-components=1` is a gnu tar option to skip the first level of directory when extracting.
In the original [install.sh](https://cnfl.io/cli) this was solved in the `execute()` function by creating a temporary directory with `mktemp`, extracting the archive there and then using the `install` command (like when using `Makefile`) to move the binary in the right place (see below snippet). I tried to keep it simpler but I can mimic the same behavior of the original script if you think that it will be cleaner, that will require just a couple more tasks to add.
```bash
  tmpdir=$(mktmpdir)
  http_download "${tmpdir}/${TARBALL}" "${TARBALL_URL}" "Accept:application/octet-stream"
# [...]
  (cd "${tmpdir}" && untar "${TARBALL}")
  install -d "${BINDIR}"
# [...]
    install "${srcdir}/${binexe}" "${BINDIR}/"
```

Reply:
I just realized that I missed to create install directory if not present, added it in a commit. That will be required anyway.